Return-Path: <clarknet!ncd.com!greg>
Return-Path: <clarknet!ncd.com!greg>
Received: from clarknet by zod.clark.net with uucp
	(Smail3.1.28.1 #2) id m0pKCU2-00003NC; Wed, 12 Jan 94 15:48 EST
Received: from welch.ncd.com by clarknet.clark.net (5.0/SMI-SVR4)
	id AA22657; Wed, 12 Jan 94 15:32:32 EST
Received: from rosebud.ncd.com (root@rosebud.ncd.com [192.43.159.161]) by welch.ncd.com (8.6.5.Beta11/8.6.4) with ESMTP id MAA18487; Wed, 12 Jan 1994 12:32:14 -0800
Received: from hansen.ncd.com (root@hansen.ncd.com [138.43.112.25]) by rosebud.ncd.com (8.6.5.Beta11/8.6.5.Beta11) with ESMTP id MAA20729 for <netaudio@ncd.com>; Wed, 12 Jan 1994 12:28:11 -0800
Received: from localhost.ncd.com (greg@localhost.ncd.com [127.0.0.1]) by hansen.ncd.com (8.6.5.Beta9/8.6.4) with SMTP id MAA10566; Wed, 12 Jan 1994 12:28:42 -0800
Message-Id: <199401122028.MAA10566@hansen.ncd.com>
X-Authentication-Warning: hansen.ncd.com: Host localhost.ncd.com didn't use HELO protocol
To: Chris Liebman <liebman@zod.xrxedds.com>
Cc: netaudio@ncd.com
Subject: Re: proplem with buckets on sparc... 
In-Reply-To: Your message of "Tue, 11 Jan 94 22:31:14 EST."
             <QhAqw2z0xSgFMO_UtL@zod.xrxedds.com> 
Date: Wed, 12 Jan 94 12:28:41 PST
From: Greg Renda <greg@ncd.com>
content-length: 1636

A little oversight when I "optimized" some code in the Sun driver.  A
signal was sneaking in at an inopportune time.  This would affect all
outputs, not just buckets.  Try this patch to server/dda/ausun.c

-Greg

===================================================================
RCS file: /u4/ncd/src/netaudio/server/dda/sun/ausun.c,v
retrieving revision 1.14
retrieving revision 1.15
diff -b -c -r1.14 -r1.15
*** 1.14	1993/12/03 01:37:44
--- 1.15	1994/01/12 20:21:57
***************
*** 19,25 ****
   * WHETHER IN AN ACTION IN CONTRACT, TORT OR NEGLIGENCE, ARISING OUT OF OR IN
   * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
   * 
!  * $NCDId: @(#)ausun.c,v 1.14 1993/12/03 01:37:44 greg Exp $
   */
  
  #include "dixstruct.h"				/* for RESTYPE */
--- 19,25 ----
   * WHETHER IN AN ACTION IN CONTRACT, TORT OR NEGLIGENCE, ARISING OUT OF OR IN
   * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
   * 
!  * $NCDId: @(#)ausun.c,v 1.15 1994/01/12 20:21:57 greg Exp $
   */
  
  #include "dixstruct.h"				/* for RESTYPE */
***************
*** 353,360 ****
--- 353,363 ----
  static void
  writeEmptyOutput()
  {
+     AuBlock         l = AuBlockAudio();
+ 
      write(devAudio, emptyOutput, 0);
      write(devAudio, emptyOutput, auMinibufSamples);
+     AuUnBlockAudio(l);
  }
  
  static void
***************
*** 375,383 ****
--- 378,391 ----
  writeOutput(p)
  AuInt16        *p;
  {
+     AuBlock         l;
+ 
      AuNativeToULAW8(p, 1, auMinibufSamples);
+ 
+     l = AuBlockAudio();
      write(devAudio, p, 0);
      write(devAudio, p, auMinibufSamples);
+     AuUnBlockAudio(l);
  }
  
  static void

